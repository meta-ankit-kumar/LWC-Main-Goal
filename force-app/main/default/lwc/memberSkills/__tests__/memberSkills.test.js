import { createElement } from 'lwc';
import MemberSkills from 'c/memberSkills';
import { createRecord } from 'lightning/uiRecordApi';
import { teamsTestData } from './data';

jest.mock('lightning/uiRecordApi', () => ({
    createRecord: jest.fn(),
}));

describe('c-member-skills', () => {
    afterEach(() => {
        while (document.body.firstChild) {
            document.body.removeChild(document.body.firstChild);
        }
    });

    it('should submit the form and handle success', async () => {
        // GIVEN
        const memberInformation = {
            Name: 'Test Member Name',
            Skills__c: 'Test skills',
            Team__c: `${teamsTestData[1].id}`
        }
        const element = createElement('c-member-skills', {
            is: MemberSkills,
        });
        element.teams = teamsTestData;
        document.body.appendChild(element);

        createRecord.mockResolvedValue({
            fields: {
                Name: { value: memberInformation.Name },
            },
        });

        // WHEN
        const form = element.shadowRoot.querySelector('form');
        const input = element.shadowRoot.querySelectorAll('input');
        const teamSelection = element.shadowRoot.querySelector('select');
        console.log('input', input);
        input[0].value = memberInformation.Name;
        input[1].value = memberInformation.Skills__c;
        teamSelection.value = memberInformation.Team__c;
        form.dispatchEvent(new CustomEvent('submit'));

        // THEN
        expect(createRecord).toHaveBeenCalledWith({
            apiName: 'TeamMember__c',
            fields: memberInformation
        });

        const wait = () => {
            return new Promise(resolve => setTimeout(resolve, 1000));
        }
        await wait();
        expect(element).toMatchSnapshot();
    });

    it('should handle form submission error', async () => {
        // GIVEN
        const element = createElement('c-member-skills', {
            is: MemberSkills,
        });
        element.teams = teamsTestData;
        document.body.appendChild(element);

        const errorMessage = 'Error message from createRecord';
        createRecord.mockRejectedValue({ message: errorMessage });
        const form = element.shadowRoot.querySelector('form');

        // WHEN
        form.dispatchEvent(new Event('submit'));

        // THEN
        const wait = () => {
            return new Promise(resolve => setTimeout(resolve, 1000));
        }
        await wait();
        expect(element).toMatchSnapshot();
    });
});
