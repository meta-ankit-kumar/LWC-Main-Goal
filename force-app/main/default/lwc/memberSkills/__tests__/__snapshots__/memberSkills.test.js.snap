// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`c-member-skills should handle form submission error 1`] = `
<c-member-skills
  x-test_memberskills-host=""
>
  #shadow-root(open)
    <div
      class="main-member-container"
      x-test_memberskills=""
    >
      <div
        class="member-skills-container"
        x-test_memberskills=""
      >
        <div
          class="error-section"
          x-test_memberskills=""
        >
          <p
            x-test_memberskills=""
          >
            There was some problem while retrieving the data: Error message from createRecord
          </p>
        </div>
      </div>
    </div>
</c-member-skills>
`;

exports[`c-member-skills should submit the form and handle success 1`] = `
<c-member-skills
  x-test_memberskills-host=""
>
  #shadow-root(open)
    <div
      class="main-member-container"
      x-test_memberskills=""
    >
      <div
        class="member-skills-container"
        x-test_memberskills=""
      >
        <div
          class="labels-container"
          x-test_memberskills=""
        >
          <label
            for="[shadow:guid]"
            x-test_memberskills=""
          >
            Name:
          </label>
          <label
            for="[shadow:guid]"
            x-test_memberskills=""
          >
            Team:
          </label>
          <label
            for="[shadow:guid]"
            x-test_memberskills=""
          >
            Skills:
          </label>
        </div>
        <div
          class="inputs-container"
          x-test_memberskills=""
        >
          <form
            x-test_memberskills=""
          >
            <input
              id="[shadow:guid]"
              minlength="4"
              name="member-name"
              required=""
              type="text"
              x-test_memberskills=""
            />
            <select
              name="member-team-name"
              required=""
              x-test_memberskills=""
            >
              <option
                value="1"
                x-test_memberskills=""
              >
                Team 1
              </option>
              <option
                value="2"
                x-test_memberskills=""
              >
                Team 2
              </option>
            </select>
            <input
              id="[shadow:guid]"
              name="member-skills"
              required=""
              type="text"
              x-test_memberskills=""
            />
            <input
              class="submit-button"
              type="submit"
              value="Submit"
              x-test_memberskills=""
            />
          </form>
        </div>
      </div>
      <div
        class="message-container"
        x-test_memberskills=""
      >
        <c-message
          x-test_memberskills=""
          x-test_message-host=""
        >
          #shadow-root(open)
            <div
              class="message-info"
              x-test_message=""
            >
              New Memeber: Test Member Name is added successfully
            </div>
        </c-message>
      </div>
    </div>
</c-member-skills>
`;
